CHANGELOG.md

```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## v1.0.2 (unreleased)
```

Comment:
Agreed.

I consider most of this to be either a tightening of the existing rules (e.g. rules for variable names) or a correction of less-than-desired behavior with respect to the underlying data types of Value objects that are produced when variable substitution is active.

The only part I'd really consider a new feature is the '$$' handling.

Any of that is enough to justify more than a patch-level version bump.